Vineyard/Vineyard/Badge.swift

```diff
@@ -6,22 +6,42 @@
 //
 
 import Foundation
+import FirebaseFirestore
 
 struct Badge: Identifiable, Codable {
-    let id: String = UUID().uuidString
-    var resolution: Resolution
-    var group: Group
+    @DocumentID var id: String? = UUID().uuidString
+    var resolutionID: String
+    var groupID: String
     var dateObtained: Date
     
+    var resolution: Resolution? = nil
+    var group: Group? = nil
+    
+    enum CodingKeys: String, CodingKey {
```

Comment:
Did you double check whether explicitly specifying the `CodingKeys` means the other properties are not encoded/decoded?